lib/mailbox.lua

```diff
@@ -3,29 +3,37 @@
 -- SPDX-License-Identifier: MIT OR GPL-2.0-only 
 --
 
-local fifo = require("fifo")
+local fifo       = require("fifo")
+local completion = require("completion")
 
 local mailbox = {} 
 local MailBox = {}
 MailBox.__index = MailBox
 
-local function new(o, allowed, forbidden)
+local function new(o, e, allowed, forbidden)
```

Comment:
Actually, I think I would keep just `o` here.. and then
```
local mbox = type(o) == 'userdata' and o or
  {queue = fifo.new(o.queue), event = completion.new(o.event)}
```
What do you think?

Reply:
it could be. then we'd have to call it like `local outbox = mailbox.outbox({queue = queue, event = event})` in dome (and, potentially, in other impls), right? it's a bit clumsy IMO, but it makes sense if we want to support native objects in the future; it could be reduced to `mailbox.outbox(inbox)`. these are two separate things albeit related, right? because impls will often have to pass them as `fifo, completion`.

Reply:
you're absolutely right.. somewhat I thought would be possible to support `mailbox.outbox(inbox)` simply by doing this.. sorry for the noise.. let's keep it as is by now..